storage/rocksdb/rdb_datadic.cc

```diff
@@ -925,13 +988,67 @@ uint Rdb_key_def::pack_record(const TABLE *const tbl, uchar *const pack_buffer,
     tuple = pack_field(field, &m_pack_info[i], tuple, packed_tuple, pack_buffer,
                        unpack_info, n_null_fields);
 
+    // Check if this column is covered.
+    if (unpack_info && m_index_type == INDEX_TYPE_SECONDARY) {
+      size_t data_length = field->data_length();
+      uint16 key_length;
+
+      // We always need the PK as a suffix, but in some cases MySQL won't add
+      // the PK suffix to a secondary key, and therefore it won't be included
+      // in key_info. If so, we need to get the key length for key parts from
+      // the PK directly.
+      DBUG_ASSERT(tbl->key_info != nullptr);
+      if (i >= tbl->key_info[get_keyno()].actual_key_parts) {
```

Comment:
:+1: It now stops writing to/reading from the bitmap when the number of columns written/read is 16. This should be fairly rare since bits are only written/read for varchars now.